src/component/plotly-graph/parse.js

```diff
@@ -120,30 +120,42 @@ function willFigureHang (result) {
     return true
   }
 
-  const numberOfPtsPerTraceType = {}
+  let maxPtBudget = 0
 
   for (let i = 0; i < data.length; i++) {
     const trace = data[i] || {}
     const type = trace.type || 'scatter'
+    const len = estimateDataLength(trace)
 
-    if (!(type in numberOfPtsPerTraceType)) {
-      numberOfPtsPerTraceType[type] = 0
-    }
+    // cap the number of points using a budget
+    maxPtBudget += len / maxPtsPerTraceType(type)
+    if (maxPtBudget > 1) return true
 
-    numberOfPtsPerTraceType[type] += estimateDataLength(trace)
+    // other special cases
 
-    // cap the number of point per trace type
-    if (numberOfPtsPerTraceType[type] > maxPtsPerTraceType(type)) {
+    // box with boxpoints: 'all'
+    if (
+      type === 'box' &&
+      trace.boxpoints === 'all' &&
+      len > 5e4
```

Comment:
In principle I guess the special cases should get included in the budget as well - otherwise you could include 199 x 5e4-point box traces...

Reply:
Good idea -> 0372d82